### PR TITLE
Refactor tax-exemption directive for reuse

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Refactor tax-exemption directive for reuse

Create factory to remove purchase factory dependency
Move checkbox out of the directive

[stage-19]

## Motivation and Context
Re-usable directive for the Subscription Manage page.

## How Has This Been Tested?
Tested locally to ensure the purchase flow tax exemption works as expected. Updated unit test coverage.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Will add the exemption to the Subscription page in a subsequent PR.

@ezequielc @santiagonoguez please review. Thanks!

File: web/partials/billing/subscription.html [hunk 1]
```diff
@@ -138,10 +138,7 @@ <h1>${{subscriptionFactory.item.subscription.plan_amount / 100 | number:2}}</h1>
         <div class="panel-body">
           <label class="mb-3">Tax Exemption</label>
           <p ng-show="subscriptionFactory.item.customer.taxability !== 'taxable'">Tax exemption applied.</p>
-          <p ng-show="subscriptionFactory.item.customer.taxability === 'taxable'">No tax exemption applied.</p>
-          <span ng-show="false">
-            <button id="addTaxExemption" class="btn btn-default btn-block" ng-click="addTaxExemption()" ng-show="subscriptionFactory.item.customer.taxability === 'taxable'">I'm Tax Exempt</button>
-          </span>
+          <button id="addTaxExemption" class="btn btn-default btn-block" ng-click="addTaxExemption()" ng-show="subscriptionFactory.item.customer.taxability === 'taxable'">I'm Tax Exempt</button>
```
Comment (santiagonoguez) [inline web/partials/billing/subscription.html:141]: just to confirm, still not implemented right ?

**Review by santiagonoguez — APPROVED** [review]
@alex-deaconu looks good to me, and seems to work well on purchase page

**Comment by ezequielc** [comment]
@alex-deaconu functionality looks good 👍

**Pull request merged**